feat: add visor docs (#6996)
vegaprotocol/vega: visor/visor-config.md

</blockquote>
</dd>

<dt>
	<code>maxNumberOfRestarts</code>  <strong>int</strong>  - optional
</dt>

<dd>

Visor starts and manages the processes of provided binaries.
This allows a user to define the maximum number of restarts in case any of
the processes have failed before the Visor process fails.

Default value: <code>3</code>

<blockquote>The amount of time Visor waits between restarts can be set by `restartsDelaySeconds`.
</blockquote>
</dd>

<dt>
	<code>restartsDelaySeconds</code>  <strong>int</strong>  - optional
</dt>

<dd>

Number of seconds that Visor waits before it tries to re-start the processes.

Default value: <code>5</code>
</dd>

<dt>
	<code>stopSignalTimeoutSeconds</code>  <strong>int</strong>  - optional
</dt>

<dd>

Number of seconds that Visor waits after it sends termination signal (SIGTERM) to running processes.
After the time has elapsed Visor force-kills (SIGKILL) any running processes.

Default value: <code>15</code>
</dd>

<dt>
	<code>upgradeFolders</code>  <strong>map[string]string</strong>  - optional
</dt>

<dd>

During the upgrade, by default Visor looks for a folder with a name identical to the upgrade version.

> The default behaviour can be changed by providing mapping between `version` and `custom_folder_name`.

Reviewer: Is this the same name that's referenced in the Fields section? If it is, this guidance on the default behaviour could be helpful to add there too.

## *VisorConfigFile*
Root of the config file

### Fields

<dl>
<dt>
	<code>maxNumberOfFirstConnectionRetries</code>  <strong>int</strong>  - optional
</dt>

<dd>

Visor communicates with Core node via RPC API. This variable allows to specify
how many times should Visor try to establish connection to Core node before the Visor process fails.
The `maxNumberOfFirstConnectionRetries` is only taken to the account
during the first start up of the Core node process - not restarts.

Default value: <code>10</code>

<blockquote>There is a 2 seconds delay between each try. Setting the max retry number to 5 means the Visor will try to establish
5 connections times in 10 seconds.
</blockquote>
</dd>

<dt>
	<code>maxNumberOfRestarts</code>  <strong>int</strong>  - optional
</dt>

<dd>

> Visor at it's core starts and manages processes of provided binaries.
> This alows to define maximum number of restarts in case that any of
> the processes has failed before the Visor process fails.

Reviewer: ```suggestion
Visor starts and manages the processes of provided binaries.
This allows a user to define the maximum number of restarts in case any of
the processes have failed before the Visor process fails.
```

Default value: <code>3</code>

> <blockquote>The amount of time Visor should wait between restarts can be set by `maxNumberOfRestarts`.

Reviewer: ```suggestion
<blockquote>The amount of time Visor waits between restarts can be set by `maxNumberOfRestarts`.
```

Reviewer: I think this description is a copy/paste of 'restartsDelaySeconds' and needs an update

</blockquote>
</dd>

<dt>
	<code>restartsDelaySeconds</code>  <strong>int</strong>  - optional
</dt>

<dd>

Number of seconds that Visor waits before it tries to re-start the processes.

Default value: <code>5</code>
</dd>

<dt>
	<code>stopSignalTimeoutSeconds</code>  <strong>int</strong>  - optional
</dt>

<dd>

> Number of seconds that Visor waits after it sends termination singal (SIGTERM) to running processes.
> After the time has elapsed the Visor force kills (SIGKILL) to running processes.

Reviewer: ```suggestion
Number of seconds that Visor waits after it sends termination signal (SIGTERM) to running processes. After the time has elapsed, Visor force-kills (SIGKILL) any running processes.
```
